packages/moxb/src/progress/ProgressImpl.ts

```diff
@@ -0,0 +1,42 @@
+import { ValueImpl, ValueOptions } from '../value/ValueImpl';
+import { Progress } from './Progress';
+import { ValueOrFunction } from '../bind/BindImpl';
+import { computed } from 'mobx';
+
+export interface ProgressOptions extends ValueOptions<ProgressImpl, number> {
+    // The active status of the progress
+    status?: ValueOrFunction<'success' | 'exception' | 'normal' | 'active'>;
+    // Progress type
+    type?: ValueOrFunction<'line' | 'circle' | 'dashboard'>;
```

Comment:
The code for defining the possible types is repeated here. (The other location is in `Progress`.)  We could define it once and use it at both locations....

Reply:
Ok, added separate type for both status and type